src/lib/tls/tls_policy.cpp

```diff
@@ -14,12 +14,18 @@
 #include <botan/tls_algos.h>
 #include <botan/tls_ciphersuite.h>
 #include <botan/tls_exceptn.h>
+#include <botan/internal/os_utils.h>
 #include <botan/internal/stl_util.h>
 #include <optional>
 #include <sstream>
 
 namespace Botan::TLS {
 
+bool Policy::allow_ssl_key_log_file() const {
+   std::string data;
+   return Botan::OS::read_env_variable(data, "SSLKEYLOGFILE");
```

Comment:
IMO this should just always return false, and then applications which use SSLKEYLOGFILE also override the policy to set it based on their needs (env variable or whatever they like). Since the application would already have to override the Callbacks to capture and format the data, this extra step does not seem to me restrictive.